src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -222,16 +239,50 @@ void YARPRobotStatePublisherModule::onRead(JointState &v)
     m_kinDynComp.setJointPos(m_jointPos);
 
     // Publish the frames on TF
-    for (size_t frameIdx=0; frameIdx < model.getNrOfFrames(); frameIdx++)
-    {
-        if(m_baseFrameIndex == frameIdx)    // skip self-tranform
-            continue;
-
-        iDynTree::Transform base_H_frame = m_kinDynComp.getRelativeTransform(m_baseFrameIndex, frameIdx);
-        iDynTree::toYarp(base_H_frame.asHomogeneousTransform(), m_buf4x4);
-        m_iframetrans->setTransform(model.getFrameName(frameIdx),
-                                    model.getFrameName(m_baseFrameIndex),
-                                    m_buf4x4);
+    bool m_publishGlobalTF = false; // TODO
+    if (m_publishGlobalTF) {
```

Comment:
I am bit confused by this if. What is this about? If the `if` is never executed, why is this included?

Reply:
Actually, this dates back to the original WIP commit by @diegoferigo (https://github.com/lrapetti/idyntree/commit/6aea5f92a83cd66d8bd0176aedc1229c30e0cb73). If it can be a useful feature, we can add the option as a possible configuration.

Reply:
@diegoferigo Any comment. 
@lrapetti If you are not using this option, I suggest to just remove it from the PR. We can always add it later.

Reply:
I have no memory about this, probably at the time I wanted to make an option in a later commit (this one was still marked as WIP).

Reply:
Ack, given that it is always disabled I think it is not necessary for the use with HDE.

Reply:
Done https://github.com/robotology/idyntree/pull/538/commits/3625d301a81976521c273c9574489553da381d43